### ISSUE TITLE
No feedback when time limit is reached during import

### ISSUE THREAD

**Issue by matthijskooijman** [LycheeOrg/Lychee-Laravel#312]
### Detailed description of the problem
I've used the import from server feature to import a big folder. This triggers the default PHP time limit (according to the PHP error log), without showing a proper error in the browser.

In the browser, I still see:

![image](https://user-images.githubusercontent.com/194491/62293572-2feae600-b469-11e9-96fb-97172adf6546.png)

This is using the current git master version (2270caf25b9f7729badaaa6f82adfae88a43de0f).

I realize that this can be solved by disabling or increasing the time limit (though perhaps having a smaller limit per image file, by calling `set_time_limit()` after each photograph, might be nicer than no limit at all), but regardless this should really provide some feedback to the user.

**Comment by matthijskooijman** [comment]
When the time limit is disabled in the config, it seems the feedback is still problematic. In the log, I find:

Thu Aug  1 14:46:30 2019 - SIGPIPE: writing to a closed pipe/socket/fd (probably the client disconnected) on request /lychee/api/Import::server (ip 217.19.19.88) !!!
Thu Aug  1 14:46:30 2019 - uwsgi_response_writev_headers_and_body_do(): Broken pipe [core/writer.c line 306] during POST /lychee/api/Import::server (217.19.19.88)
[pid: 15248|app: -1|req: -1/11711] 217.19.19.88 () {52 vars in 1964 bytes} [Thu Aug  1 14:37:53 2019] POST /lychee/api/Import::server => generated 0 bytes in 517226 msecs (HTTP/1.1 200) 13 headers in 0 bytes (0 switches on core 0)

And the browser is still stuck in the same screen. Looking at the browser net panel, it seems the request returned a 504 Gateway Timeout, so this might be nginx that cuts off the connection (60s seems to be default timeout). This is probably more of an installation issue (it seems that Lychee could help here by sending some data every now and then to reset the [nginx uwsgi timeout](http://nginx.org/en/docs/http/ngx_http_uwsgi_module.html#uwsgi_read_timeout), but I'm not sure if that would survive all kinds of buffering). 

It does seem all photographs were imported properly, though.

I guess the nicest solution would be progress reporting or even some kind of background processing queue, but for now a working error message would be nice :-)

**Comment by kamil4** [comment]
Just yesterday I was diagnosing a memory leak in _Input from Server_ which turned out to be primarily due to... debugging being enabled. When I disabled debugging (`APP_DEBUG=false` in `.env`) it started working much better for me. Before, the server would run out of memory before 200 pictures could be imported; after, I succeeded importing over 1000. How many were you trying to import?

In general, the whole import functionality is tricky for us... Unlike the regular upload, where we issue a series of short-lived requests, one for each file, with import it's a single request for everything, and it's all handled in one thread on the server, so execution time limits become an issue, as do memory leaks and now apparently also client-side timeouts...

Are you sure there was no error message in the browser when execution time limit got exceeded? I got one when I ran out of memory. It's easy to miss though because it's only displayed for a few seconds at the top of the window (and it would've been a generic "Server error or API not found"). Those errors are also normally logged in the JS console, so you can check there...

I agree that providing a progress report would be nice, especially if it would reset the timeout counter as a bonus... We don't have any such functionality in the server at the moment though to draw an inspiration from. @ildyria, I wonder if we could create a `StreamedResponse` and periodically output some custom-formatted status message that way, to be read by an `onprogress` callback in the front end?

**Comment by matthijskooijman** [comment]
> Just yesterday I was diagnosing a memory leak in Input from Server which turned out to be primarily due to... debugging being enabled.

Oh, excellent point, I still have that enabled too. This was with some 150 images in one go, but it's also a fairly slow Atom-based server. I tried again with debugging enabled, but a 126-file folder still took a few minutes.

> Are you sure there was no error message in the browser when execution time limit got exceeded? I got one when I ran out of memory. It's easy to miss though because it's only displayed for a few seconds at the top of the window 

Oh, that could have been. I was not watching the window, since it was not very interesting at the time :-p. But even if there was a short error popup, the "progress" dialog (showing "importing") should be closed, or perhaps changed to to "failed" or something then, and maybe have the actual error message stick around longer?

I just tried again and did watch the window at the time the request failed, but did not see any message now (the failing request is `/api/Import::server`).

As for progress feedback: Maybe send the list of scanned files to the client and show those in the browser, making the process more similar to the multi-file-upload UI?
